general/device.hpp

```diff
@@ -273,6 +273,18 @@ class Device
    */
    static inline MemoryType GetDeviceMemoryType() { return Get().device_mem_type; }
 
+   static inline MemoryType GetDeviceTemporaryMemoryType()
```

Comment:
This type should be user configurable, e.g. users may want to use Umpire quick pool (setup as `MemoryType::DEVICE_UMPIRE_2`) for temporaries.

Reply:
Yes, that's the plan. Should it just be a static function in `Device` like `Device::SetDeviceTemporaryMemoryType`?

Also, `GetDeviceTemporaryMemoryType` seems verbose. Is there a shorter alternative that you would prefer?

Reply:
> Yes, that's the plan. Should it just be a static function in `Device` like `Device::SetDeviceTemporaryMemoryType`?

The user configuration for the memory types is now done with `Device::SetMemoryTypes`. One option is to add a version of `Device::SetMemoryTypes` with 4 args -- 2 new args for host-temp-type and device-temp-type. Another option is to add a new method `Device::SetTempMemoryTypes`. In either case, the 2-argument `Device::SetMemoryTypes` should set defaults for the temp-types that make sense. Also, we should check if any relevant changes are needed in `Device::UpdateMemoryTypeAndClass` and in `MemoryManager::Configure`.

> Also, `GetDeviceTemporaryMemoryType` seems verbose. Is there a shorter alternative that you would prefer?

The only thing that comes to mind is to shorten `Temporary` to `Temp`: `GetDeviceTempMemoryType`.